grails-app/services/au/org/ala/ecodata/MetadataService.groovy

```diff
@@ -135,8 +135,34 @@ class MetadataService {
         // Remove deprecated activities
         activities = activities.findAll {!it.status || it.status == ACTIVE}
 
-        Map byCategory = [:]
+        groupActivities(activities)
+    }
+
+    /**
+     * Returns a Map of activity types grouped by category.
+     * @param programId Most of the legacy projects are updated recently to insert programId (see github #2741)
+     * @return a Map, key: String, value: List of name, description for each activity in the category
+     */
+    Map activitiesListByProgramId(String programId) {
+        Map config = Program.findByProgramId(programId)?.config
+        List<String> activities = config?.activities.collect{it.name}
 
+        List forms = activityFormService.search(category:activities)
+        forms = forms.findAll{it.publicationStatus == PublicationStatus.PUBLISHED}
```

Comment:
Searching for published forms is probably a good idea here, but you should pass it as a criteria rather than doing post-filtering.